src/condor/htcondor_helper.py

```diff
@@ -2,174 +2,332 @@
+                    
+                    # Check for timeout on held jobs
+                    if total_wait >= 120:
+                        job_is_idle = status_counts.get(1, 0) > 0
+                        job_is_running = status_counts.get(2, 0) > 0
+                        if not job_is_idle and not job_is_running:
+                            raise Exception(f"Cluster {cluster_id} timed out (job was held too long)")
+
+                    time.sleep(check_interval)
+                    total_wait += check_interval
+                    
+                except Exception as query_error:
+                    self.logger.warning(f"Query error: {query_error}")
+                    time.sleep(check_interval)
+                    total_wait += check_interval
 
         except Exception as e:
-            logger.error(f"Error monitoring jobs: {e}")
+            self.logger.error(f"Error monitoring jobs: {e}")
             raise
 
     def retrieve_output(self, cluster_id: int):
         """Retrieve output files from completed jobs"""
-        
         try:
-            self._add_credentials()
+            # Add credentials for required services
+            if self.credd:
+                for service in ["rdrive", "scitokens"]:
+                    try:
+                        self.credd.add_user_service_cred(htcondor.CredType.OAuth, b"", service)
+                    except Exception as e:
+                        self.logger.warning(f"Could not add credential for {service}: {e}")
+            
             self.schedd.retrieve(f"ClusterId == {cluster_id}")
-            logger.info(f"Successfully retrieved output for cluster {cluster_id}")
+            self.logger.info(f"Successfully retrieved output for cluster {cluster_id}")
         except Exception as e:
-            logger.error(f"Failed to retrieve output: {e}")
+            self.logger.error(f"Failed to retrieve output: {e}")
             raise
 
-    def _add_credentials(self):
-        """Add credentials for required services"""
-        if self.credd:
-            for service in ["rdrive", "scitokens"]:
-                try:
-                    self.credd.add_user_service_cred(htcondor.CredType.OAuth, b"", service)
-                except Exception as e:
-                    print(f"Could not add credential for {service}: {e}")
\ No newline at end of file
+    def cleanup(self, cluster_id: int):
+        """Clean up after job completion"""
+        try:
+            # Add credentials for required services
+            if self.credd:
+                for service in ["rdrive", "scitokens"]:
+                    try:
+                        self.credd.add_user_service_cred(htcondor.CredType.OAuth, b"", service)
+                    except Exception as e:
+                        self.logger.warning(f"Could not add credential for {service}: {e}")
+            
+            # Remove all jobs from the queue
```

Comment:
do we want to remove all jobs from the queue? if a job is held do we want to inspect it? or do we just want to rely on our logs?